src/scripts/main.js

```diff
@@ -1,3 +1,219 @@
 'use strict';
 
-// write your code here
+const score = document.querySelector('.game-score');
+const button = document.querySelector('.button');
+const cells = document.querySelector('tbody');
+const startGameText = document.querySelector('.message-start');
+const loseGameText = document.querySelector('.message-lose');
+const winGameText = document.querySelector('.message-win');
+
+const numOfCells = 4;
+let scoreCount = 0;
+let gameField = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+button.addEventListener('click', () => {
+  button.classList.replace('start', 'restart');
+  button.innerText = 'Restart';
+  startGameText.classList.add('hidden');
+  loseGameText.classList.add('hidden');
+  winGameText.classList.add('hidden');
+
+  startPlay();
+});
+
+function hasEmptyTile() {
+  for (let r = 0; r < numOfCells; r++) {
+    if (gameField[r].includes(0)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+function setRundom() {
+  while (hasEmptyTile()) {
+    const randomRow = Math.floor((Math.random() * numOfCells));
+    const randomCol = Math.floor((Math.random() * numOfCells));
+
+    if (gameField[randomRow][randomCol] === 0) {
+      const numb = Math.random() < 0.9 ? 2 : 4;
+
+      gameField[randomRow][randomCol] = numb;
+      break;
+    }
+  }
+
+  setCells();
+}
+
+function startPlay() {
+  gameField = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  scoreCount = 0;
+  score.innerText = scoreCount;
+
+  setRundom();
+  setRundom();
+}
+
+function loseGame() {
+  if (hasEmptyTile()) {
+    return false;
+  }
+
+  for (let r = 0; r < numOfCells; r++) {
+    for (let c = 0; c < numOfCells; c++) {
+      if (gameField[r][c] === gameField[r][c + 1]) {
+        return false;
+      }
+    }
+  }
+
+  for (let r = 0; r < numOfCells - 1; r++) {
+    for (let c = 0; c < numOfCells; c++) {
+      if (gameField[r][c] === gameField[r + 1][c]) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+function setCells() {
+  for (let r = 0; r < numOfCells; r++) {
+    for (let c = 0; c < numOfCells; c++) {
+      const currentCell = cells.rows[r].cells[c];
+      const num = gameField[r][c];
+
+      currentCell.innerText = '';
+      currentCell.classList.value = '';
+      currentCell.classList.add('field-cell');
+
+      if (num > 0) {
+        currentCell.innerText = num;
+        currentCell.classList.add(`field-cell--${num}`);
+      }
+
+      if (num === 2048) {
+        winGameText.classList.remove('hidden');
+        button.classList.replace('restart', 'start');
+      }
+    }
+  }
+
+  if (loseGame()) {
+    loseGameText.classList.remove('hidden');
+  }
+}
+
+document.addEventListener('keyup', (e) => {
```

Comment:
arrow buttons work before I click "Start":
![image](https://github.com/mate-academy/js_2048_game/assets/86923089/6e819628-37e4-4acb-93c5-2154ba1bf1d9)

add this listener on the "Start" button click (i.e. inside another listener). Define the function separately that you use here